web/src/connectors/edit-model-connector/edit-model-connector.tsx

```diff
@@ -0,0 +1,195 @@
+        setStepIndex(stepIndex + 1);
+    };
+    const handlePrev = () => {
+        setStepIndex(stepIndex - 1);
+    };
+    const finishButtonCaption = 'Add Model';
+
+    const renderForm = useCallback(
+        ({ lens, save }: IFormApi<ModelValues>) => {
+            const steps: WizardPropsStep[] = [
+                {
+                    title: 'Model',
+                    content: (
+                        <>
+                            <div className={wizardStyles['content__body']}>
+                                <EditModelSettings
+                                    lens={lens}
+                                    categories={categories?.data}
+                                    basements={basements?.data}
+                                    models={models?.data}
+                                />
+                            </div>
+                            <div className={wizardStyles['content__footer']}>
+                                {renderWizardButtons({
+                                    onNextClick: handleNext
+                                })}
+                            </div>
+                        </>
+                    )
+                },
+                {
+                    title: 'Data',
+                    content: (
+                        <>
+                            <div className={wizardStyles['content__body']}>
+                                <EditModelData lens={lens} />
+                            </div>
+                            <div className={wizardStyles['content__footer']}>
+                                {renderWizardButtons({
+                                    onPreviousClick: handlePrev,
+                                    onNextClick: () => {
+                                        save();
+                                    },
+                                    nextButtonCaption: finishButtonCaption
+                                })}
+                            </div>
+                        </>
+                    )
+                }
+            ];
+            return <Wizard steps={steps} returnUrl={MODELS_PAGE} stepIndex={stepIndex} />;
+        },
+        [stepIndex, basements]
+    );
+
+    const editModelMutation = useEditModelMutation();
+
+    const handleOnModelEdit = useCallback(async (values: ModelValues) => {
+        let categories: (string | number)[] = [];
+        if (values.jobs) {
+            values.jobs.forEach((el) => {
+                categories = [...categories, ...el.categories];
+            });
+        } else {
+            values.categories?.forEach((el) => {
+                categories = [...categories, el.name];
+            });
+        }
+        categories.map((el) => {
+            return el.toString();
+        });
+        const model: Model = {
+            id: values.id,
+            name: values.name || '',
+            basement: values.basement?.id || '',
+            categories: categories as string[],
```

Comment:
Maybe we can define `categories` as `string[]` at the beginning? To avoid `as string[]`